nautobot/extras/signals.py

```diff
@@ -455,6 +456,22 @@ def job_result_delete_associated_files(instance, **kwargs):
             file_proxy.file.delete()
 
 
+@receiver(m2m_changed, sender=JobModel)
+def add_default_job_queue_to_job_queues(instance, **kwargs):
+    # Add specified default_job_queue to job.job_queues if it is not included.
+    default_job_queue = instance.default_job_queue
+    action = kwargs.pop("action", None)
+    if action in ["post_remove"]:
+        instance.job_queues.add(default_job_queue)
```

Comment:
Perhaps this needs to be something like:

```suggestion
def add_default_job_queue_to_job_queues(instance, action, model, pk_set, **kwargs):
    if action == "pre_remove":
        if model is JobModel:  # job_model.job_queues.remove()
            # Don't allow removing the default job queue
            pk_set.discard(instance.default_job_queue.pk)
        elif model is JobQueue:  # job_queue.jobs.remove()
            # Don't allow removing jobs that this queue is default for
            for job_model in instance.default_for_jobs.all():
                pk_set.discard(job_model.pk)
    elif action == "post_clear":
        if model is JobModel:  # job_model.job_queues.clear()
            # Re-add the default job queue
            instance.job_queues.add(instance.default_job_queue)
        elif model is JobQueue:  # job_queue.jobs.clear()
            # Re-add the jobs this queue is default for
            for job_model in instance.default_for_jobs.all():
                job_model.job_queues.add(instance.pk)
```